tensorflow/compiler/tf2tensorrt/convert/ops/layer_utils.h

```diff
@@ -0,0 +1,365 @@
+      nvinfer1::ITensor* false_values) noexcept {
+    TRT_ENSURE(network_);
+    TRT_ENSURE(bool_condition);
+    TRT_ENSURE(true_values);
+    TRT_ENSURE(false_values);
+    nvinfer1::ISelectLayer* layer =
+        network_->addSelect(*bool_condition, *true_values, *false_values);
+    TRT_ENSURE(layer);
+    return layer;
+  }
+
+  // Simulates the creation of a tensor of the same shape as "input", but filled
+  // with 1's of type int32 where "input" is non-zero, otherwise it is zero.
+  // TensorRT doesn't have a "cast" operation, so this simulates with a
+  // "select/where" operation.
+  StatusOr<nvinfer1::ISelectLayer*> NonZeroInt(
+      nvinfer1::ITensor* input) noexcept {
+    TRT_ENSURE(network_);
+    TRT_ENSURE(input);
+    StatusOr<nvinfer1::IConstantLayer*> zeros =
+        this->Constant<int32>(0, input->getDimensions().nbDims);
+    TRT_ENSURE_PTR_OK(zeros);
+    StatusOr<nvinfer1::IConstantLayer*> ones =
+        this->Constant<int32>(1, input->getDimensions().nbDims);
+    TRT_ENSURE_PTR_OK(ones);
+    StatusOr<nvinfer1::IElementWiseLayer*> equal_to_zero_mask =
+        this->Equal(input, (*zeros)->getOutput(0));
+    TRT_ENSURE_PTR_OK(equal_to_zero_mask);
+    return this->Where((*equal_to_zero_mask)->getOutput(0), input,
+                       (*ones)->getOutput(0));
+  }
+
+  // Adds an elementwise multiplication operation to the network.
+  StatusOr<nvinfer1::IElementWiseLayer*> Mul(nvinfer1::ITensor* lhs,
+                                             nvinfer1::ITensor* rhs) noexcept {
+    TRT_ENSURE(network_);
+    TRT_ENSURE(lhs);
+    TRT_ENSURE(rhs);
+    nvinfer1::IElementWiseLayer* layer = network_->addElementWise(
+        *lhs, *rhs, nvinfer1::ElementWiseOperation::kPROD);
+    TRT_ENSURE(layer);
+    return layer;
+  }
+
+  // Adds a Constant layer whose output is a TensorRT shape tensor. The shape
+  // tensor's size and values correspond to dim's nbDims and d[], respectively.
+  StatusOr<nvinfer1::IConstantLayer*> Constant(
+      const nvinfer1::Dims& shape_data) noexcept {
+    TRT_ENSURE(network_);
+    TRT_ENSURE(shape_data.nbDims > 0);
+    nvinfer1::Dims shape_dims;
+    shape_dims.nbDims = 1;
+    shape_dims.d[0] = shape_data.nbDims;
+    TRT_ShapedWeights const_weights =
+        weight_store_->GetTempWeights(nvinfer1::DataType::kINT32, shape_dims);
+    int32* values = const_weights.GetPointer<int32>();
+    for (int i = 0; i < shape_data.nbDims; i++) {
+      values[i] = static_cast<int32>(shape_data.d[i]);
+    }
+    nvinfer1::IConstantLayer* const_layer = network_->addConstant(
+        const_weights.shape_, const_weights.GetTrtWeights());
+    TRT_ENSURE(const_layer);
+    nvinfer1::ITensor* output = const_layer->getOutput(0);
+    TRT_ENSURE(output);
+    TRT_ENSURE(output->getType() == nvinfer1::DataType::kINT32);
+    TRT_ENSURE(const_layer);
+    return const_layer;
+  }
+
+  // Adds a Constant layer composed of a single layer
```

Comment:
Incomplete comment?
Let's use this:
Adds a Constant layer that produces a tensor with a single value "scalar". The tensor has "nb_dims" dimensions and each dimension has only one element. The data type of the tensor is determined by the data type of "scalar".

Reply:
fixed